### ISSUE TITLE
Error in log_warning() : could not find function "log_warning"

### ISSUE THREAD

**Issue by TatyanaLev** [landerlabcode/BigSurR#1]
Hey Kai,

BigSur is not running for me because the log_warning function appears to be missing.  How can I correct this?

I tried with correlations = T and correlations = F, and the error is the same both ways.

Thanks!

epi_bigsur <- BigSur(epi, variable.feature=T, correlations=T)

[1] "Pipeline started execution."
Error in log_warning() : could not find function "log_warning"
[1] "Error in log_warning() : could not find function \"log_warning\"\n"
[1] "Traceback:"
 [1] "10: doWithOneRestart(return(expr), restart)"                                            
 [2] "9: withOneRestart(expr, restarts[[1L]])"                                                
 [3] "8: withRestarts({"                                                                      
 [4] "       .Internal(.signalCondition(simpleWarning(msg, call), msg, "                      
 [5] "           call))"                                                                      
 [6] "       .Internal(.dfltWarn(msg, call))"                                                 
 [7] "   }, muffleWarning = function() NULL)"                                                 
 [8] "7: .signalSimpleWarning(\"sparse->dense coercion: allocating vector of size 7.0 GiB\", "
 [9] "       base::quote(asMethod(object)))"                                                  
[10] "6: asMethod(object)"                                                                    
[11] "5: as(x, \"matrix\")"                                                                   
[12] "4: as.matrix.Matrix(GetAssayData(object = seuratob, slot = counts.slot))"               
[13] "3: as.matrix(GetAssayData(object = seuratob, slot = counts.slot))"                      
[14] "2: get.residuals(seurat.obj, assay, counts.slot, c, depthlist)"                         
[15] "1: BigSur(epi, variable.feature = T, correlations = T)"

sessionInfo()
R version 4.2.2 (2022-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Rocky Linux 8.8 (Green Obsidian)

Matrix products: default
BLAS/LAPACK: /opt/apps/OpenBLAS/0.3.21/lib/libopenblas_zenp-r0.3.21.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] BigSur_1.0.1       Seurat_5.0.3       SeuratObject_5.0.1 sp_2.1-3          

loaded via a namespace (and not attached):
  [1] Rtsne_0.17             colorspace_2.1-0       deldir_2.0-4          
  [4] ellipsis_0.3.2         ggridges_0.5.6         RcppHNSW_0.6.0        
  [7] spatstat.data_3.0-4    leiden_0.4.3.1         listenv_0.9.1         
 [10] ggrepel_0.9.5          RSpectra_0.16-1        fansi_1.0.6           
 [13] codetools_0.2-18       splines_4.2.2          rootSolve_1.8.2.4     
 [16] polyclip_1.10-6        spam_2.10-0            jsonlite_1.8.8        
 [19] common_1.1.2           ica_1.0-3              cluster_2.1.4         
 [22] png_0.1-8              uwot_0.1.16            shiny_1.8.0           
 [25] sctransform_0.4.1      spatstat.sparse_3.0-3  compiler_4.2.2        
 [28] httr_1.4.7             Matrix_1.6-5           fastmap_1.1.1         
 [31] lazyeval_0.2.2         cli_3.6.2              later_1.3.2           
 [34] htmltools_0.5.7        tools_4.2.2            igraph_2.0.3          
 [37] dotCall64_1.1-1        gtable_0.3.4           glue_1.7.0            
 [40] RANN_2.6.1             reshape2_1.4.4         dplyr_1.1.4           
 [43] Rcpp_1.0.12            scattermore_1.2        vctrs_0.6.5           
 [46] spatstat.explore_3.2-7 nlme_3.1-160           progressr_0.14.0      
 [49] lmtest_0.9-40          spatstat.random_3.2-3  stringr_1.5.1         
 [52] globals_0.16.3         mime_0.12              miniUI_0.1.1.1        
 [55] lifecycle_1.0.4        irlba_2.3.5.1          goftest_1.2-3         
 [58] future_1.33.1          MASS_7.3-58.1          zoo_1.8-12            
 [61] scales_1.3.0           promises_1.2.1         spatstat.utils_3.0-4  
 [64] parallel_4.2.2         RColorBrewer_1.1-3     reticulate_1.35.0     
 [67] pbapply_1.7-2          gridExtra_2.3          ggplot2_3.5.0         
 [70] stringi_1.8.3          fastDummies_1.7.3      logr_1.3.6            
 [73] rlang_1.1.3            pkgconfig_2.0.3        moments_0.14.1        
 [76] matrixStats_1.2.0      pracma_2.4.4           lattice_0.20-45       
 [79] ROCR_1.0-11            purrr_1.0.2            tensor_1.5            
 [82] patchwork_1.2.0        htmlwidgets_1.6.4      cowplot_1.1.3         
 [85] tidyselect_1.2.1       parallelly_1.37.1      RcppAnnoy_0.0.22      
 [88] plyr_1.8.9             magrittr_2.0.3         R6_2.5.1              
 [91] generics_0.1.3         withr_3.0.0            pillar_1.9.0          
 [94] fitdistrplus_1.1-11    survival_3.4-0         abind_1.4-5           
 [97] tibble_3.2.1           future.apply_1.11.1    KernSmooth_2.23-20    
[100] utf8_1.2.4             spatstat.geom_3.2-9    plotly_4.10.4         
[103] grid_4.2.2             data.table_1.15.2      digest_0.6.35         
[106] xtable_1.8-4           tidyr_1.3.1            httpuv_1.6.14         
[109] munsell_0.5.0          viridisLite_0.4.2

**Comment by silksmoothie** [comment]
Wow I'm sorry Tatyana I didn't see this until now. I will check on this and get back to you.

**Comment by ogandril** [comment]
Has this been corrected? I also have the same issue here.

> adre.output <- BigSur(adre, variable.feature=T, correlations=F)
[1] "Pipeline started execution."
Error in log_warning() : could not find function "log_warning"
[1] "Error in log_warning() : could not find function \"log_warning\"\n"
[1] "Traceback:"
 [1] "10: doWithOneRestart(return(expr), restart)"                                             
 [2] "9: withOneRestart(expr, restarts[[1L]])"                                                 
 [3] "8: withRestarts({"                                                                       
 [4] "       .Internal(.signalCondition(simpleWarning(msg, call), msg, "                       
 [5] "           call))"                                                                       
 [6] "       .Internal(.dfltWarn(msg, call))"                                                  
 [7] "   }, muffleWarning = function() NULL)"                                                  
 [8] "7: .signalSimpleWarning(\"sparse->dense coercion: allocating vector of size 13.9 GiB\", "
 [9] "       base::quote(asMethod(object)))"                                                   
[10] "6: asMethod(object)"                                                                     
[11] "5: as(x, \"matrix\")"                                                                    
[12] "4: as.matrix.Matrix(GetAssayData(object = seuratob, slot = counts.slot))"                
[13] "3: as.matrix(GetAssayData(object = seuratob, slot = counts.slot))"                       
[14] "2: get.residuals(seurat.obj, assay, counts.slot, c, depthlist)"                          
[15] "1: BigSur(adre, variable.feature = T, correlations = F)"    

When I try setting log.file = F as a parameter, then it starts computing but then R crashes (no message)and I have to start it again from scratch.

**Comment by silksmoothie** [comment]
Hi Olivier. I fix for this will be out soon. For now, please continue performing your analyses with log.file = F. This parameter will not affect your results and is purely to track the progression of analysis.

**Comment by ogandril** [comment]
The problem is that it leads to an R session abortion in that case (image attached)....

<img width="910" alt="Capture d’écran 2024-11-27 à 07 54 53" src="https://github.com/user-attachments/assets/93319c67-ab10-4afb-916a-190ca6b0e4a1">

**Comment by silksmoothie** [comment]
To me, that appears to be a separate issue. Often I've experienced R sessions aborting when the amount of RAM required to run a process greatly exceeds the amount available. I see in the console something mentioning sparse to dense matrix conversion, can you please send me the entire kernel output you get from that? Thanks for your patience while I sort this out.

**Comment by silksmoothie** [comment]
Unfortunately, that issue could also be caused by things outside of my direct control -  the solution could be as simple as restarting your R session or re-downloading BigSur, or much more complicated, but I will do my best to look into it.

**Comment by ogandril** [comment]
> To me, that appears to be a separate issue. Often I've experienced R sessions aborting when the amount of RAM required to run a process greatly exceeds the amount available. I see in the console something mentioning sparse to dense matrix conversion, can you please send me the entire kernel output you get from that? Thanks for your patience while I sort this out.

Could that be due to the size of the matrix? (here I am trying to analyse a 5.5 GB seurat matrix).
Not sure where I can find the "kernel output"?

Thank's for your help!

PS: I tried to force BigSur re-installation but to no avail.

**Comment by silksmoothie** [comment]
It's possible that it is due to the size of the matrix taking up too much RAM on your device. As a test, I recommend trying on a small subset of your data (i.e., 100 cells and 100 genes) to see if it completes on that small set. After that, we can discuss potential fixes to prune the data further without losing too much information. And sorry, I meant the output that you see in the "console" section of R studio after you run the code.

**Comment by ogandril** [comment]
I tried 1000 genes and 1000 cells. After figuring ou how to remove null genes I obtained:

> adre3
An object of class Seurat 
941 features across 1000 samples within 1 assay 
Active assay: RNA (941 features, 69 variable features)
 3 layers present: counts, data, scale.data
 2 dimensional reductions calculated: pca, umap

I now can run BigSur with no log.file error, nor R crashing, so this was indeed a size issue!
That a problem since I would like to make a feauture selection with my complete matrix.

But I also now have a new problem

> adre.output <- BigSur(adre3, variable.feature=T, correlations=F)
[1] "Pipeline started execution."
[1] "Pre-computed depthlist was either not supplied or supplied as a non-numeric vector. Calculating depth scaling from scratch."
[1] "Modified corrected Pearson residuals calculated."
[1] "Modified corrected Fano factors calculated."
[1] "Beginning identification of significant mcFanos."
Error in (function (cl, name, valueClass)  : 
  ‘meta.data’ is not a slot in class “Assay”
[1] "Error in (function (cl, name, valueClass)  : \n  ‘meta.data’ is not a slot in class “Assay”\n"
[1] "Traceback:"
 [1] "3: stop(gettextf(\"%s is not a slot in class %s\", sQuote(name), dQuote(cl)), "                                                                     
 [2] "       domain = NA)"                                                                                                                                
 [3] "2: (function (cl, name, valueClass) "                                                                                                               
 [4] "   {"                                                                                                                                               
 [5] "       ClassDef <- getClass(cl)"                                                                                                                    
 [6] "       slotClass <- ClassDef@slots[[name]]"                                                                                                         
 [7] "       if (is.null(slotClass)) "                                                                                                                    
 [8] "           stop(gettextf(\"%s is not a slot in class %s\", sQuote(name), "                                                                          
 [9] "               dQuote(cl)), domain = NA)"                                                                                                           
[10] "       if (.identC(slotClass, valueClass)) "                                                                                                        
[11] "           return(TRUE)"                                                                                                                            
[12] "       ok <- possibleExtends(valueClass, slotClass, ClassDef2 = getClassDef(slotClass, "                                                            
[13] "           where = .classEnv(ClassDef)))"                                                                                                           
[14] "       if (isFALSE(ok)) "                                                                                                                           
[15] "           stop(gettextf(\"assignment of an object of class %s is not valid for @%s in an object of class %s; is(value, \\\"%s\\\") is not TRUE\", "
[16] "               dQuote(valueClass), sQuote(name), dQuote(cl), slotClass), "                                                                          
[17] "               domain = NA)"                                                                                                                        
[18] "       TRUE"                                                                                                                                        
[19] "   })(\"Assay\", \"meta.data\", \"data.frame\")"                                                                                                    
[20] "1: BigSur(adre3, variable.feature = T, correlations = F)"   

If I run:
> slotNames(adre@assays$RNA)
[1] "counts"        "data"          "scale.data"    "assay.orig"    "var.features"  "meta.features"
[7] "misc"          "key"     

There indeed seems not to be a meta.data class here...

Thank's a lot for your patience.

**Comment by silksmoothie** [comment]
It appears like you might have an outdated version of Seurat. The "meta.features" slot has been changed to "meta.data" in versions 5 and up. I recommend checking to see that you have the most updated version, though I should plan to add some backwards compatibility with the older versions. 

I have removed the dependency of BigSur on the logr package, so the original errors should no longer occur. I will be closing this issue. Olivier, I would be happy to continue helping you to troubleshoot the package - and, if you were looking to run the program on some larger data, I would be happy to run it for you on my device. Feel free to reach out to me at ksilkwoo@uci.edu and we can continue this conversation!
